### DISCUSSION TITLE
Fix author in text citation in CSL file

### DISCUSSION THREAD

**Discussion by rgaiacs** [quarto-dev/quarto-cli#2612] [Q&A]
Consider [mwe.zip](https://github.com/quarto-dev/quarto-cli/files/9645931/mwe.zip)

The minimal working example Markdown file is

```
---
title: MWE
---

Foo @rezende_cric_2021 foo

Bar [@rezende_cric_2021].
```

when I run `pandoc --from markdown --citeproc --bibliography=references.bib --csl=ieee.csl mwe.md --to native`, I get

```
[ Para
    [ Str "Foo"
    , Space
    , Cite
        [ Citation
            { citationId = "rezende_cric_2021"
            , citationPrefix = []
            , citationSuffix = []
            , citationMode = AuthorInText
            , citationNoteNum = 1
            , citationHash = 0
            }
        ]
        [ Str "[1]" ]
    , Space
    , Str "foo"
    ]
, Para
    [ Str "Bar"
    , Space
    , Cite
        [ Citation
            { citationId = "rezende_cric_2021"
            , citationPrefix = []
            , citationSuffix = []
            , citationMode = NormalCitation
            , citationNoteNum = 2
            , citationHash = 0
            }
        ]
        [ Str "[1]" ]
    , Str "."
    ]
```

The `citationMode` is correct. `@rezende_cric_2021` is `AuthorInText` and `[@rezende_cric_2021]` is `NormalCitation`.

When I run `pandoc --from markdown --citeproc --bibliography=references.bib --csl=ieee.csl mwe.md --to html`, I get something like

> <p>Foo <span class="citation" data-cites="rezende_cric_2021">[1]</span> foo</p>
> <p>Bar <span class="citation" data-cites="rezende_cric_2021">[1]</span>.</p>

instead of

> <p>Foo <span class="citation" data-cites="rezende_cric_2021">T. Rezende <em>et al.</em> (2021)</span> foo</p>
> <p>Bar <span class="citation" data-cites="rezende_cric_2021">[1]</span>.</p>

Can I have some help fixing the CSL?

**Comment by cderv** [comment]
Isn't `ieee.csl` a Numeric style ? This is what I found in https://www.zotero.org/styles?q=id%3Aieee and what is inside the CSL
````
<category citation-format="numeric"/>
````

From specification, this is for `[1]` citation, not author date style (https://docs.citationstyles.org/en/stable/specification.html#info)

P12 of IEEE editorial guide shows example of numeric: http://journals.ieeeauthorcenter.ieee.org/wp-content/uploads/sites/7/IEEE-Editorial-Style-Manual-for-Authors.pdf

Why did you expect the other form with this CSL file ?

**Reply by rgaiacs** [reply]
I want to be able to use [BibLaTeX](https://ctan.org/pkg/biblatex?lang=en)'s [text commands](https://mirror-hk.koddos.net/CTAN/macros/latex/contrib/biblatex/doc/biblatex.pdf), for example `\citeauthor`. This is a long feature request to Pandoc, see https://github.com/jgm/pandoc/issues/2335 and https://github.com/jgm/pandoc/issues/7258.

Citation Style Language (CSL) defines five types of style:

- "author-date",  (Doe, 1999);
- "author", (Doe);
- "numeric", [1];
- "label", [doe99]; and
- "note", ¹.

My expectation is that Pandoc will use the CSL type for `NormalCitation` (e.g. `[@doe]`) and provide a appropriated version for `AuthorInText` (e.g `@doe`). This works for "author-date" but not for the others.

**Reply by cderv** [reply]
I don't know if I fully understand. here is my understanding of your comments. 

https://github.com/jgm/pandoc/issues/2335 is about support some new LaTeX commands in the Pandoc LaTeX reader when convertex from TeX file. 

For https://github.com/jgm/pandoc/issues/7258, it is indeed about Markdown command for some style on the reference. Quarto mechanism is based on [Pandoc citation](https://pandoc.org/MANUAL.html#citations) processing and we support the same syntax. (https://quarto.org/docs/authoring/footnotes-and-citations.html#sec-citations) So when Pandoc will support the command, we would too.

But still I am not sure we understand the same thing on CSL. 

First, CSL will be used only with Pandoc citation processing, not biblatex or natbib. Then, CSL defines the style, and there is several categories. `ieee.csl`, as I shared in above link to zotero, is numeric style. So I don't think you can get the author in there, it looks like that in text
![image](https://user-images.githubusercontent.com/6791940/192725261-85be3ea7-3eba-4c31-b849-796f07a53a15.png)

> This works for "author-date" but not for the others.

It works for this because the CSL style allows this, not the other. 

This is my understanding

**Reply by rgaiacs** [reply]
@cderv Thanks for your time to answer. I will bring this up with Pandoc developers.

**Reply by cderv** [reply]
Cool. Do not hesitate to share the issue link or pandoc discussion link when you open one with them. Thank you

**Reply by rgaiacs** [reply]
@cderv I created https://github.com/jgm/pandoc/issues/8331 to start a discussion among the Pandoc community.
